### ISSUE TITLE
Android: no serializer found for class dd.a and no properties discovered to create BeanSerializer

### ISSUE THREAD

**Issue by Luk3s** [optimizely/optimizely-flutter-sdk#47]
The plugin works fine on iOS and in the Android emulator.
When I build the APK and install it on a real device I get a `PlatformException`.

This is the exact exception I see in the logs
```
PlatformException(error, No serializer found for class dd.a and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["experimentflag"]), null, java.lang.IllegalArgumentException: No serializer found for class dd.a and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.HashMap["experimentflag"])
 	at x3.t.f(SourceFile:21)
 	at x3.t.r(SourceFile:1)
 	at dd.i.r(SourceFile:13)
 	at dd.j.onMethodCall(SourceFile:19)
 	at pe.j$a.a(SourceFile:2)
 	at ce.c.j(SourceFile:2)
 	at ce.c.k(SourceFile:2)
 	at ce.c.g(Unknown Source:0)
 	at ce.b.run(Unknown Source:12)
 	at android.os.Handler.handleCallback(Handler.java:942)
 	at android.os.Handler.dispatchMessage(Handler.java:99)
 	at android.os.Looper.loopOnce(Looper.java
 #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:653)
 #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:296)
 <asynchronous suspension>
 #2      OptimizelyUserContext._decide (package:optimizely_flutter_sdk/src/user_context/optimizely_user_context.dart:145)
 <asynchronous suspension>
 #3      OptimizelyUserContext.decideForKeys (package:optimizely_flutter_sdk/src/user_context/optimizely_user_context.dart:125)
...
```

From `flutter doctor`
```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.2 22D49 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
```

From `build.gradle`
```
compileSdkVersion 33
minSdkVersion 21
targetSdkVersion 33
```

Let me know if there is anything else I can provide.

**Comment by russell-loube-optimizely** [comment]
Hi @Luk3s , thank you for raising this. We're taking a look as a team on this issue later today and I will provide any updates from our initial investigation.

**Comment by TannerLessen** [comment]
FYI for anyone else affected as our team was and find this page.. this issue was fixed by [this PR](https://github.com/optimizely/optimizely-flutter-sdk/pull/51) and released in [v1.0.1](https://github.com/optimizely/optimizely-flutter-sdk/releases/tag/v1.0.1) though the versioning is unclear
